### PR TITLE
fix: missing interfaces and edge cases

### PR DESCRIPTION
**Pull request by jalberto-solink**
### Fixes

The reason for this PR is to handle additional address edge cases after a list of different current addresses were provided

### Details
  - Added additional address types to ensure existing parsing is maintained
- Added support for:
  - State highways
  - State Routes
  - US Highways
  - Noonlight zip plus four
  - Street directionals N S E W
- Refactored address parser to use helper functions to help with readability

### tests
- Seperated tests per country - Canada, US, Puerto Rico
- Added additional edge case tests that were previously failing

### CODE REVIEW

File: utils/utils.js [hunk 1]
```diff
@@ -6,4 +6,61 @@ export function toTitleCase(str) {
...
+        Object.keys(streetTypes).join('|') + ')\\b\\.?' + 
+        '( +(?:' + streetDirectionalString + ')\\b)?', 'i');
+}
+
+export function handleAddressPart(addressParts, stateString) {
+    if (stateString.length > 0) {
+        addressParts[addressParts.length-1] = stateString;
+        return addressParts[addressParts.length-1];
+    } else {
+        addressParts.splice(-1,1);
```
[the remark this answers is not in the document]
Comment (jalberto-solink) [reply utils/utils.js:30]: This was fixed

**Review by gerryburtenshaw — APPROVED** [review]
